tests/test_csr.py

```diff
@@ -139,3 +139,115 @@ def test_parse_csr2(self):
+                util.OrderedDict([
+                    ('type', 'microsoft_request_client_info'),
+                    (
+                        'values',
+                        [
+                            util.OrderedDict([
+                                ('clientid', 5),
+                                ('machinename', 'windows.host.example.net'),
+                                ('username', 'locuser'),
+                                ('processname', 'MMC.EXE'),
+                            ])
+                        ])
+                    ]
+                ),
+                util.OrderedDict([
+                    ('type', 'microsoft_enrollment_csp_provider'),
+                    (
+                        'values',
+                        [
+                            util.OrderedDict([
+                                ('keyspec', 1),
+                                ('cspname', 'Microsoft RSA SChannel Cryptographic Provider'),
+                                ('signature', ()),
+                            ])
+                        ]
+                    ),
+                ]),
+                util.OrderedDict([
+                    ('type', 'extension_request'),
+                    (
+                        'values',
+                        [
+                            [
+                                util.OrderedDict([
+                                    ('extn_id', 'microsoft_enroll_certtype'),
+                                    ('critical', False),
+                                    (
+                                        'extn_value',
+                                        'Machine',
+                                    )
+                                ]),
+                                util.OrderedDict([
+                                    ('extn_id', 'extended_key_usage'),
+                                    ('critical', False),
+                                    (
+                                        'extn_value',
+                                        ['client_auth', 'server_auth'],
+                                    ),
+                                ]),
+                                util.OrderedDict([
+                                    ('extn_id', 'key_usage'),
+                                    ('critical', False),
+                                    (
+                                        'extn_value',
+                                        set(['digital_signature', 'key_encipherment']),
+                                    ),
+                                ]),
+                                util.OrderedDict([
+                                    ('extn_id', 'key_identifier'),
+                                    ('critical', False),
+                                    (
+                                        'extn_value',
+                                        bytes.fromhex('2a 98 4b c1 ff 6e 16 ed 2d 69 35 0a 26 e7 1f 8c 05 4f b8 e6'),
```

Comment:
Since asn1crypto supports 2.6+, this will need to be changed to something like:

```
b'\x2a\x98\x4b\xc1\xff\x6e\x16\xed\x2d\x69\x35\x0a\x26\xe7\x1f\x8c\x05\x4f\xb8\xe6',
```

Reply:
Ah, so bytearray.fromhex is out too?

Very impressed by your response time by the way. :zap:

Reply:
It looks like `bytearray.fromhex()` was in at least 2.7, but then you'd have to cast/convert to a byte string anyway.

Reply:
It looks like the test suites pass, but if you'd prefer it i'll change it to a byte literal anyway, it was just an attempt to make it more easily comparable to the output of `openssl req -noout -text -inform der -in tests/fixtures/test-windows-host.csr`

Reply:
This was my kids computer…